pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -816,6 +816,27 @@ public void run() {
               //       CONSUMING -> ONLINE state transition.
               segmentLock.lockInterruptibly();
               try {
+                // TODO(akkhanch): this has been added here as we want to create a new consuming segment for the
+                //  partition
+                //  With the new commit protocol we might not need this as the segment consumed itself will take care
+                //  of the next segment commit start
+                // TODO (akkhanch): keep this after the lock has been acquired as we don't want the state transition
+                //  to impact this build. The state transition from CONSUMING -> ONLINE will also try to acquire lock
+                //  impacting this build and ingestions
+
+                // TODO: might need to support the two ways in which stream config can be set. On under the ingestion
+                //  config and other under the tableIndexConfig
+                if (isPauselessEnabeld()) {
```

Comment:
We want to Initiate the COMMIT_START before building the segment. This is crucial for allowing ingestion while the current segment is being built.